meson.build

```diff
@@ -0,0 +1,236 @@
+project('WPEBackend-fdo', ['c', 'cpp'],
+	meson_version: '>=0.49',
+	default_options: [
+		'b_ndebug=if-release',
+		'c_std=c99',
+		'cpp_eh=none',
+		'cpp_std=c++11',
+	],
+	license: 'BSD-2-Clause',
+	version: '1.5.0',
+)
+
+# This refers to the API level provided. This is modified only with major,
+# breaking changes, which need modifications in programs using the library
+# before they can be compiled again.
+api_version = '1.0'
+
+# Before making a release, the LT_VERSION string should be modified.
+# The string is of the form [C, R, A].
+# - If interfaces have been changed or added, but binary compatibility has
+#   been preserved, change to [C+1, 0, A+1].
+# - If binary compatibility has been broken (eg removed or changed interfaces)
+#   change to [C+1, 0, 0]
+# - If the interface is the same as the previous version, use [C, R+1, A].
+soversion = [6, 0, 5]
+
+# Split the *project* version into its components.
+version_info = meson.project_version().split('.')
+version_info = {
+	'PROJECT_VERSION_MAJOR': version_info[0],
+	'PROJECT_VERSION_MINOR': version_info[1],
+	'PROJECT_VERSION_PATCH': version_info[2],
+}
+
+# Mangle [C, R, A] into an actual usable *soversion*.
+soversion_major = soversion[0] - soversion[2]  # Current-Age
+soversion_minor = soversion[2]  # Age
+soversion_micro = soversion[1]  # Revision
+soversion = '@0@.@1@.@2@'.format(soversion_major, soversion_minor, soversion_micro)
+
+sources = [
+	'src/exported-image-egl.cpp',
+	'src/fdo.cpp',
+	'src/initialize-egl.cpp',
+	'src/ipc.cpp',
+	'src/renderer-backend-egl.cpp',
+	'src/renderer-host.cpp',
+	'src/view-backend-exportable-fdo.cpp',
+	'src/view-backend-exportable-fdo-egl.cpp',
+	'src/view-backend-exportable-private.cpp',
+	'src/ws-client.cpp',
+	'src/ws.cpp',
+	'src/linux-dmabuf/linux-dmabuf.cpp',
+	'src/linux-dmabuf/linux-dmabuf-protocol.c',
+]
+
+api_headers = [
+	'include/wpe-fdo/exported-image-egl.h',
+	'include/wpe-fdo/initialize-egl.h',
+	'include/wpe-fdo/view-backend-exportable.h',
+	'include/wpe-fdo/fdo-egl.h',
+	'include/wpe-fdo/fdo.h',
+
+	# Generated API headers.
+	configure_file(
+		input: 'include/wpe-fdo/version.h.cmake',
```

Comment:
Ah, this could be renamed as `version.h.in` instead of using the `.cmake` suffix; I'll upload a new revision with that change applied.